tests/rbe_repo/BUILD

```diff
@@ -103,6 +103,24 @@ sh_test(
     ],
 )
 
+sh_test(
```

Comment:
@smukherj1 Do you have any additional types of tests you would want to see? We just duplicated some basic cases that demonstrated running rbe_autoconfig on Windows behaves as expected for these general cases.

Should we try to run all the different types of tests on Windows that test the different parameters or do you think this is sufficient for now?

Reply:
I'll defer suggesting additional test types back to you :)

Rationale: My knowledge of rbe_autoconfig is fairly high level because I'm not the original author. I think at this point you have a better understanding of its internals than me so are in a better position to suggest tests for additional coverage. Overall, we would like to avoid breaking you when we make any updates to rbe_autoconfig. We don't have any major features planned for rbe_autoconfig so this should hopefully be a rare occurrence. Maybe we'll realize the holes in our testing if such a breakage does happen.

One thing I am concerned though is regarding the runtime of the new windows tests. I know windows base images are huge, orders of GB vs the 100-200 MB range for Linux images. Will have to see how long these run on the CI and if they hit timeouts or run too long, we might have to re-evaluate our windows testing strategy.

Reply:
Makes sense, we can add an integration type test like in https://github.com/bazelbuild/bazel-toolchains/blob/e7cd66eccd2532c7601fbac3c6f93088cb5b4153/tests/rbe_repo/output_base.yaml. Having that integration test is probably going to help us the most in terms of breakages I would guess

Regarding the container image size, yeah that is definitely true. An image with VC++ installed from VS Build Tools is on the order of 10GB and the base windows 2019 image is ~5GB. Does the bazel CI service have docker installed on the workers that run Windows? Does GCB have Windows workers? If they have docker and windows container base images cached, it could make pulling these images much faster

To distill it into a summary of the different CI/remote execution systems at play:
- bazel CI/buildkite - I believe currently supports Windows but needs docker installed and enough disk space to pull Windows images locally
- GCB - does it support executing tests on Windows?
- RBE - it supports Windows but we would need a worker pool to execute integration tests in for this project

Reply:
I don't think GCB has Windows workers. Bazel CI seems to support windows but I don't know if they have docker pre-installed. Probably need to try it out and see.

I think we should first aim to get the basic config generation and file existence tests up and running before we get into integration tests. The former will require setting up an environment capable of downloading a Windows docker image and generating configs for it. Once that's sorted out, it shouldn't be too much additional effort to launch an RBE build.

Reply:
@smukherj1 Looks like Bazel CI (at least this naive attempt at using it, see the changes to the presubmit pipeline) does not have Docker installed: https://buildkite.com/bazel/bazel-toolchains/builds/14514#d7eaa2b6-98f2-480c-8580-63001451bc86, unless it is hiding somewhere in the environment and we can set `DOCKER_PATH` to the cli client

Reply:
Does it fail even if you specify docker_tool_path as `docker.exe`? I wonder if this is just because of ctx.which not working on Windows or not working with `docker` instead of `docker.exe` [here](https://github.com/bazelbuild/bazel-toolchains/blob/fffc0403b097ebff4bb5259ec6eb8694df0ee545/rules/rbe_repo/util.bzl#L148)

@philwo Do you know if the buildkite Windows workers have docker installed?

Reply:
We found https://github.com/bazelbuild/continuous-integration/blob/00dd8eb623c36204a26d97cd6236a67898ea0539/buildkite/setup-windows.ps1 which implies that they do not have Docker installed

We can PR to that repo, but if the Windows image this software is being installed on does not have the `Containers` feature installed, installing Docker is not super useful as we won't be able to run a container

Reply:
(I don't recall if installing the Containers feature requires a reboot, if not, we can PR that too)

Reply:
@smukherj1 We don't install Docker on our Windows machines. We also have no experience in the team with using / setting up Docker on Windows, so I would like to avoid installing it.

I saw above that you could possibly test via RBE if we had a Windows worker pool - maybe that's easier to get?

Reply:
@philwo we're trying to test a repository rule that downloads a windows container and runs Bazel inside it to generate toolchain configs. Bazel would never send the script executed in the repository rule to RBE so the `Downloading the docker image` and `Running bazel inside the downloaded Docker image` need to be supported on the host.

Actually we're just looking for any windows CI capable of spinning up docker containers. GCB doesn't support windows so we were trying to figure out if we could use Buildkite.

Reply:
@sunjayBhatia It's possible in the short term you might have to proceed without adding CI coverage for Windows in bazel-toolchains :(

I can take an action to add testing for this next quarter. Please do let us know if we appear to be breaking you frequently (once a month or more) and I'll be able to prioritize adding the test. Given that it's proving to be challenging to set up windows presubmit on Github, I might just set up a nightly test that runs windows toolchain config generation using bazel-toolchains HEAD.

Reply:
@smukherj1 sounds good to me as a stepping stone, we can add the tests we've written here and just disable them on Linux so they can be turned on in the future in Windows CI